cmd/kube-apiserver/app/server.go

```diff
@@ -632,23 +633,28 @@ func BuildStorageFactory(s *options.ServerRunOptions, apiResourceConfig *servers
 	return storageFactory, nil
 }
 
-// DefaultOptions set default ServerRunOptions.
+// preparedServerRunOptions is a private wrapper that enforces a call of PrepareRunOptions() before Run can be invoked.
+type preparedServerRunOptions struct {
+	*options.ServerRunOptions
+}
+
+// PrepareRunOptions set default ServerRunOptions.
 // Should be called after kube-apiserver flags parsed.
-func DefaultOptions(s *options.ServerRunOptions) error {
+func PrepareRunOptions(s *options.ServerRunOptions) (*preparedServerRunOptions, error) {
```

Comment:
`Prepare` or `Complete`? I tend to `Complete`.  Gives more context what it does.

Reply:
SGTM, I could hardly imagine a name.